2018/003-offline-chart-repository.md

```diff
@@ -0,0 +1,441 @@
+
+For example, given the following directory `/home/flavio/suse-charts`:
+
+```
+$ ls -l /home/flavio/suse-charts
+total 388
+-rw-r--r-- 1 flavio users 89627 Jul  3 10:00 cf-2.0.2.tgz
+-rw-r--r-- 1 flavio users 32344 Jul  3 10:00 cf-2.10.1.tgz
+-rw-r--r-- 1 flavio users 24001 Jul  3 10:00 cf-2.4.1-beta3+cf278.0.g12f5d3a8.tgz
+-rw-r--r-- 1 flavio users 20567 Jul  3 10:00 cf-2.5.0-beta4+cf278.0.gafa3d0e9.tgz
+-rw-r--r-- 1 flavio users 21575 Jul  3 10:00 cf-2.6.11.tgz
+-rw-r--r-- 1 flavio users 21589 Jul  3 10:00 cf-2.6.1-rc1+cf278.0.g52d7a644.tgz
+-rw-r--r-- 1 flavio users 21645 Jul  3 10:00 cf-2.7.0.tgz
+-rw-r--r-- 1 flavio users 26401 Jul  3 10:00 cf-2.8.0.tgz
+-rw-r--r-- 1 flavio users  2268 Jul  3 10:00 cf-usb-sidecar-mysql-0.0.0-366-gcf3d373.tgz
+-rw-r--r-- 1 flavio users  2271 Jul  3 10:00 cf-usb-sidecar-mysql-0.0.0-pre.35-develop.tgz
+-rw-r--r-- 1 flavio users  2257 Jul  3 10:00 cf-usb-sidecar-mysql-1.0.1.tgz
+-rw-r--r-- 1 flavio users  2500 Jul  3 10:00 cf-usb-sidecar-postgres-0.0.0-366-gcf3d373.tgz
+-rw-r--r-- 1 flavio users  2507 Jul  3 10:00 cf-usb-sidecar-postgres-0.0.0-pre.23-develop.tgz
+-rw-r--r-- 1 flavio users  2489 Jul  3 10:00 cf-usb-sidecar-postgres-1.0.1.tgz
+-rw-r--r-- 1 flavio users  4467 Jul  3 10:00 console-0.9.7.tgz
+-rw-r--r-- 1 flavio users  3878 Jul  3 10:00 console-0.9.8.tgz
+-rw-r--r-- 1 flavio users  3880 Jul  3 10:00 console-0.9.9.tgz
+-rw-r--r-- 1 flavio users  3871 Jul  3 10:00 console-1.0.0.tgz
+-rw-r--r-- 1 flavio users  3873 Jul  3 10:00 console-1.0.2.tgz
+-rw-r--r-- 1 flavio users  4078 Jul  3 10:00 console-1.1.0.tgz
+-rw-r--r-- 1 flavio users 11243 Jul  3 10:00 uaa-2.0.2.tgz
+-rw-r--r-- 1 flavio users  6180 Jul  3 10:00 uaa-2.10.1.tgz
+-rw-r--r-- 1 flavio users  6988 Jul  3 10:00 uaa-2.4.1-beta3+cf278.0.g12f5d3a8.tgz
+-rw-r--r-- 1 flavio users  4152 Jul  3 10:00 uaa-2.5.0-beta4+cf278.0.gafa3d0e9.tgz
+-rw-r--r-- 1 flavio users  4143 Jul  3 10:00 uaa-2.6.11.tgz
+-rw-r--r-- 1 flavio users  4175 Jul  3 10:00 uaa-2.6.1-rc1+cf278.0.g52d7a644.tgz
+-rw-r--r-- 1 flavio users  4144 Jul  3 10:00 uaa-2.7.0.tgz
+-rw-r--r-- 1 flavio users  5470 Jul  3 10:00 uaa-2.8.0.tgz
+```
+
+A simple read-only chart registry can be created with the following command:
+
+```
+docker run --rm \
+  -v /home/flavio/charts:/charts:ro \
+  -p 4000:4000 \
+  chartmuseum/chartmuseum \
+    --port 4000 \
+    --disable-api \
+    --storage local \
+    --storage-local-rootdir /charts
+```
+The above command mounts the charts directory from the host into the container
+using the read-only mode. Then it starts ChartMuseum with the following flags:
+
+* `--disable-api`: all the write operations (create, update, delete) are
+  disabled, only read operations are allowed.
+* `--storage` and `--storage-local-rootdir`: all the charts are stored on the
+  local filesystem and are inside of the `/charts` directory.
+
+The `index.yaml` file can be obtained by executing: `curl http://localhost:4000`.
```

Comment:
Remove this instruction as it is not needed. Unless this is meant as a learning step?